src/modules/src/error_estimator_ukf.c

```diff
@@ -0,0 +1,1623 @@
+{
+  uint8_t ii, jj, kk;
+  float L[DIM_FILTER][DIM_FILTER] = {0};
+
+  cholesky(&covNavFilter[0][0], &L[0][0], DIM_FILTER);
+
+  for (jj = 0; jj < (DIM_FILTER + 2); jj++)
+  {
+    for (ii = 0; ii < DIM_FILTER; ii++)
+    {
+      sigmaPoints[ii][jj] = 0.0f;
+    }
+  }
+  // Compute actual sigma points taking into account current estimate and covariance
+  for (jj = 0; jj < (DIM_FILTER + 2); jj++)
+  {
+    for (ii = 0; ii < DIM_FILTER; ii++)
+    {
+
+      for (kk = 0; kk < DIM_FILTER; kk++)
+      {
+        sigmaPoints[ii][jj] = sigmaPoints[ii][jj] + L[ii][kk] * sigmaPointsTempl[kk][jj];
+      }
+      sigmaPoints[ii][jj] = sigmaPoints[ii][jj] + xEst[ii];
+    }
+  }
+}
+
+static bool ukfUpdate(float *Pxy, float *Pyy, float innovation)
+{
+  uint8_t ii, jj;
+  float Kk[DIM_FILTER] = {0};
+  float covNew[DIM_FILTER][DIM_FILTER] = {0};
+  float KkRKkTp[DIM_FILTER][DIM_FILTER] = {0};
+  bool doneUpdate = false;
+
+  for (ii = 0; ii < DIM_FILTER; ii++)
+  {
+    for (jj = 0; jj < DIM_FILTER; jj++)
+    {
+      covNew[ii][jj] = covNavFilter[ii][jj];
+    }
+  }
+
+  for (ii = 0; ii < DIM_FILTER; ii++)
+  {
+    Kk[ii] = Pxy[ii] / Pyy[0];
+    xEst[ii] = xEst[ii] + Kk[ii] * innovation;
+  }
+  for (ii = 0; ii < DIM_FILTER; ii++)
+  {
+    for (jj = 0; jj < DIM_FILTER; jj++)
+    {
+      KkRKkTp[ii][jj] = Kk[ii] * Kk[jj] * Pyy[0];
+      covNew[ii][jj] = covNew[ii][jj] - KkRKkTp[ii][jj];
+    }
+  }
+  for (ii = 0; ii < DIM_FILTER; ii++)
+  {
+    for (jj = 0; jj < DIM_FILTER; jj++)
+    {
+      covNavFilter[ii][jj] = 0.5f * covNew[ii][jj] + 0.5f * covNew[jj][ii];
+    }
+  }
+
+  if (useNavigationFilter)
+  {
+    resetNavigationStates();
+    doneUpdate = true;
+  }
+  return doneUpdate;
+}
+
+// reset strapdown navigation after filter update step if measurements were obtained
+static void resetNavigationStates()
+{
+  uint8_t ii;
+
+  float attVec[3] = {xEst[6], xEst[7], xEst[8]};
+  float quatNav[4] = {stateNav[6], stateNav[7], stateNav[8], stateNav[9]};
+  float attQuat[4], quatRes[4];
+
+  if ((isnan(xEst[0])) || (isnan(xEst[1])) ||
+      (isnan(xEst[2])) || (isnan(xEst[3])) ||
+      (isnan(xEst[4])) || (isnan(xEst[5])) ||
+      (isnan(xEst[6])) || (isnan(xEst[7])) ||
+      (isnan(xEst[8])))
+  {
+    for (ii = 0; ii < DIM_FILTER; ii++)
+    {
+      xEst[ii] = 0.0f;
+    }
+    nanCounterFilter++;
+  }
+  else
+  {
+    // update position and velocity state
+    for (ii = 0; ii < 6; ii++)
+    {
+      stateNav[ii] = stateNav[ii] + xEst[ii];
+    }
+
+    quatFromAtt(&attVec[0], &attQuat[0]);
+    multQuat(&quatNav[0], &attQuat[0], &quatRes[0]);
+
+    for (ii = 6; ii < 10; ii++)
+    {
+      stateNav[ii] = quatRes[ii - 6];
+    }
+
+    for (ii = 0; ii < DIM_FILTER; ii++)
+    {
+      xEst[ii] = 0.0f;
+    }
+
+    directionCosineMatrix(&quatRes[0], &dcm[0][0]);
+    transposeMatrix(&dcm[0][0], &dcmTp[0][0]);
+    computeSigmaPoints();
+  }
+}
+
+static uint8_t cholesky(float *A, float *L, uint8_t n)
```

Comment:
There is a similar function in the tdoa robust measurement model... so actually we should separate these type of functions. https://github.com/bitcraze/crazyflie-firmware/blob/af96c070d6bc32cf73eeb89980a970d7e01f7f8c/src/modules/src/kalman_core/mm_tdoa_robust.c#L25-L27

This might be out of the scope of this PR for now perhaps but for us for any refactoring in the future.  I've made an ticket to start collecting those I've seen so far #1182